general_rules/ExternalComputing.tex

```diff
@@ -26,6 +26,34 @@ \section{External computing}\label{rule:robot_external_computing}
 
 \textbf{Remark:} Teams are allowed to use their own software in the external computing devices (not only cloud services). This software must be publicly available to other teams for scientific purposes (evaluation, test, and benchmarking), as well as for TC for inspection. Although open-sourcing the software is not mandatory, this practice is advised and encouraged by the league.
 
+\subsection{External Computing Devices on site for SSPL (Hardware)}
+\label{rule:robot_external_computing_devices_sspl}
+
+\begin{itemize}
+  \item \textbf{Location:} all external computing devices, approved by the \iaterm{Technical Committee}{TC}
+  		(see~\ref{rule:robot_external_computing}), must be placed at a designated location close to the arena.
+  		The location features a table (or otherwise suitable furnishing) and an ethernet switch (wired)
+        that connects to the arena WIFI. In the remainder of this document, this location is called
+  		\textbf{E}xternal \textbf{C}omputing \textbf{R}esource \textbf{A}rea (ECRA).
+  \item \textbf{Procedure:} before a teams' test run, external comp. devices must be setup in time.
+  		Immediately after a teams' test slot, all equipment must be removed from the ECRA in order to give ensuing
+        teams adequate space and time to setup their devices. No team should ever occupy the ECRA continuously.
+  		As soon as the referee indicates the teams' test slot starts, team members are strictly forbidden to touch
+        their external comp. devices until the current test run is over. Breaking this ``hands-off'' rule
+        is penalized with 0 (zero) points for the corresponding test. Using external comp. devices during the test
+        from any other location than the ECRA will be penalized with \textbf{at least 0 (zero) points for the test}.
+  \item \textbf{Limitations:} external computing devices currently have no limitation concerning computation
+  		power and form factor. However, in order to maintain enough space and a	clear arrangement (e.g. wrt
+        inspectability for referees), stand-alone screens are \textbf{not allowed}. Mice and keyboards connected
+        to external computing devices are allowed.
+  \item \textbf{Recommendations:} show up in time and setup up your devices early enough. If your team is up last,
+        do not setup first. Consider the ECRA a FIFO queue. There will be enough space at the ECRA to establish a
+        buffer (of teams), so if you show up in time and everyone removes their equipment the transition will be smooth.
+\end{itemize}
+
+The referees, Technical Committee, and Organizing Committee members may run random checks anytime during the competition
```

Comment:
Add a \testbf{Remark:} at the beginning of the paragraph. A reference to the Fair Play rule is necessary as well as specifying that the team will be disqualified (is stronger that mentioning just a zero-point penalty) for the test, or even from the entire competition.

Reply:
I'm not sure that I like the idea of having the compute devices moving while people are in line. That could really slow up the line going through. Think of how long it could potentially take to start an external device, start all of the software, and so forth. I'd rather teams leave their gear hooked up and simply not communicate with the device and/or start their software until they're about to compete, but I think that moving the physical devices around could be chaotic.

Reply:
Me neither, but having the devices of all 12 teams can also be problematic. Do you have in mind any alternative solution?

Reply:
I'm not sure. The way this is stated really makes problems, though. There shouldn't be an external monitor, meaning that the external device either must be a laptop or permit someone to run SSH or a terminal emulator into the system. Plus you have boot times to think of. Consider how crazy it is when the ref is going, "<Team Name>, it's time to set up your robot and compete or forfeit your turn." With this setup, teams may still be booting up their hardware and wrestling with network problems during that time. I really think that we just need something like a long table, not inside the lane where the teams are going, and then a requirement that you unplug your network cable until it's your turn. This allows teams to get their computers going without much issue. Maybe we give them a max footprint for their hardware on the table. Everything must fit inside this square.